### ISSUE TITLE
App Randomly Triggers Exceptions Between Page Loads

### ISSUE THREAD

**Issue by jt-kl** [DevToys-app/DevToys#1194]
### Current behavior

Experiencing "soft" exceptions which seems to be happening randomly when I transition between tools. To be clear, these exceptions are being caught and the following message is displayed at the bottom of the app: "An unhandled error has occured. Reload"

Clicking the "Reload" link will reload the app but when switching between tool pages, the error will re-occur. 

### How to reproduce it (as minimally and precisely as possible)

**Environment Information**:
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.4 LTS
Release:	22.04
Codename:	jammy

Steps To Reproduce
- Launch the application
- Randomly select any tools from any of the major categories: Converters, Encoders/Decoders, Formatters and etc. (FYI, I'm not using the "quick shortcuts/history" links)

### Expected behavior

Doesn't crash.

### Affected platforms

Linux

### Affected DevToys kind

DevToys (app with GUI)

### DevToys Version

2.0-preview.1

### Relevant Assets/Logs

[log-20240612T000125.txt](https://github.com/user-attachments/files/15791986/log-20240612T000125.txt)

**Comment by karkcoder** [comment]
Can confirm same issue on PopOS. Its very random. Is there a way to dump log on what's throwing exception?
![Screenshot from 2024-06-11 22-19-20](https://github.com/DevToys-app/DevToys/assets/9093473/ae983baf-35af-44b9-98e7-55eb192e2f0f)

**Comment by veler** [comment]
Thanks for this information. I'm not familiar with PopOS. On Linux (at least Debian), logs are stored in `home/.local/share/devtoys/Logs/`. The folder `home/.local/share` may change based on the value of the `XDG_DATA_HOME` environment variable, if you have one.

**Comment by karkcoder** [comment]
Found the log however log is all INFO only, there is no error logs. App crashed right after JsonWebTokenEncoderDecoder was loaded.

![Screenshot from 2024-06-11 22-43-45](https://github.com/DevToys-app/DevToys/assets/9093473/68452d47-d8c1-4a9b-ad1e-6cd80da15dfa)


`2024-06-11T22:41:37.5006399-06:00	Information	[DevToys.Blazor.Core.Services.FontService]	[LogImportThirdPartyFontsAsync]	Loaded 2 font(s) in 21.0786 ms and injected them in 301.446 ms
2024-06-11T22:41:39.3611345-06:00	Information	[DevToys.Core.Tools.GuiToolProvider]	[LogSetMostRecentUsedTool]	Set 'Base64ImageEncoderDecoder' as the most recently used tool.
2024-06-11T22:41:39.3621636-06:00	Information	[DevToys.Core.Tools.GuiToolInstance]	[LogInstanceCreated]	Instance of 'Base64ImageEncoderDecoder' tool created.
2024-06-11T22:41:41.1571385-06:00	Information	[DevToys.Core.Settings.SettingsProvider]	[LogSetSetting]	Setting 'DevToys.Core.RecentTool2' changed to 'Base64ImageEncoderDecoder'
2024-06-11T22:41:41.1574107-06:00	Information	[DevToys.Core.Settings.SettingsProvider]	[LogSetSetting]	Setting 'DevToys.Core.RecentTool1' changed to 'Settings'
2024-06-11T22:41:41.1574334-06:00	Information	[DevToys.Core.Tools.GuiToolProvider]	[LogSetMostRecentUsedTool]	Set 'Settings' as the most recently used tool.
2024-06-11T22:41:41.1587355-06:00	Information	[DevToys.Core.Tools.GuiToolInstance]	[LogInstanceCreated]	Instance of 'Settings' tool created.
2024-06-11T22:42:00.8325654-06:00	Information	[DevToys.Core.Settings.SettingsProvider]	[LogSetSetting]	Setting 'DevToys.Core.RecentTool3' changed to 'Base64ImageEncoderDecoder'
2024-06-11T22:42:00.8333391-06:00	Information	[DevToys.Core.Settings.SettingsProvider]	[LogSetSetting]	Setting 'DevToys.Core.RecentTool2' changed to 'Settings'
2024-06-11T22:42:00.8333590-06:00	Information	[DevToys.Core.Settings.SettingsProvider]	[LogSetSetting]	Setting 'DevToys.Core.RecentTool1' changed to 'JsonWebTokenEncoderDecoder'
2024-06-11T22:42:00.8333744-06:00	Information	[DevToys.Core.Tools.GuiToolProvider]	[LogSetMostRecentUsedTool]	Set 'JsonWebTokenEncoderDecoder' as the most recently used tool.
2024-06-11T22:42:00.8399005-06:00	Information	[DevToys.Core.Tools.GuiToolInstance]	[LogInstanceCreated]	Instance of 'JsonWebTokenEncoderDecoder' tool created.`

**Comment by tyrossel** [comment]
I also have this message when switching tools on PopOS. I have those errors logs:

```
JAVASCRIPT Exception caught:
MESSAGE: JS object instance with ID 2 does not exist (has it been disposed?).
w@app://localhost/_framework/blazor.webview.js:1:5282
@app://localhost/_framework/blazor.webview.js:1:3045
Promise@[native code]
beginInvokeJSFromDotNet@app://localhost/_framework/blazor.webview.js:1:3018
@app://localhost/_framework/blazor.webview.js:1:48025
@user-script:1:4:79
forEach@[native code]
@user-script:1:4:49
global code@app://localhost/:1:26
STACK TRACE:
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at Microsoft.JSInterop.JSObjectReferenceExtensions.InvokeVoidAsync(IJSObjectReference jsObjectReference, String identifier, Object[] args)
   at DevToys.Blazor.Core.Extensions.IJSRuntimeExtensions.InvokeVoidWithErrorHandlingAsync(IJSObjectReference jsObjectReference, String identifier, Object[] args) in /home/etienne/RiderProjects/Publish/submodules/DevToys/src/app/dev/DevToys.Blazor/Core/Extensions/IJSRuntimeExtensions.cs:line 66
```

**Comment by olivierwph** [comment]
i have the same problem in linux mint.

when i open devtoys, i have the log 
beginInvokeJSFromDotNet@app://localhost/_framework/blazor.webview.js:1:3018
@app://localhost/_framework/blazor.webview.js:1:48025
@user-script:1:4:79
forEach@[native code]
@user-script:1:4:49
global code@app://localhost/:1:26
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
   at DevToys.Blazor.Core.Extensions.IJSRuntimeExtensions.InvokeVoidWithErrorHandlingAsync(IJSRuntime jsRuntime, String identifier, Object[] args) in /home/etienne/RiderProjects/Publish/submodules/DevToys/src/app/dev/DevToys.Blazor/Core/Extensions/IJSRuntimeExtensions.cs:line 23

then when i click on any tool (in menu), i have 
"an unhandled error has occured. Reload"
no addition log !

**Comment by SaifAqqad** [comment]
Getting the same behavior but with a different error message when navigating between pages

![image](https://github.com/DevToys-app/DevToys/assets/47293197/bcc21757-4e09-49b1-bf93-762e2b4f9617)

```log
Received unexpected acknowledgement for render batch 295 (next batch should be 294)
   at Microsoft.AspNetCore.Components.WebView.Services.WebViewRenderer.NotifyRenderCompleted(Int64 batchId)
   at Microsoft.AspNetCore.Components.WebView.IpcReceiver.OnMessageReceivedAsync(PageContext pageContext, String message)
   at Microsoft.AspNetCore.Components.WebView.WebViewManager.<>c__DisplayClass18_0.<<MessageReceived>b__0>d.MoveNext()
```

I'm running Linux Mint 21.3 (which is based on Ubuntu 22.04)

**Comment by veler** [comment]
Hey everyone,
Thanks for your feedback here. I just published an update of DevToys, but it doesn't address this issue yet. 
The issue you're facing is now my top priority. I will look at it closely in the next few days. Thank you in advance for your patience and sorry for the inconvenience.

**Comment by veler** [comment]
Hello,
I think I have a fix: https://github.com/DevToys-app/DevToys/pull/1259
At least, I do not reproduce the issue anymore since this change :D

**Comment by veler** [comment]
Hi,
I published a new update of DevToys. Please try it out and let me know if the problem still reproduce. Thanks :)

**Comment by jt-kl** [comment]
My preliminary tests of 2.0-preview.3 indicates the problem has been resolved; at least for Ubuntu 22.04.4 LTS. Thank you

**Comment by karkcoder** [comment]
Can confirm, its working fine in Pop!_OS 22.04 LTS now. Thank you for the fix!!
